packages/angular/cli/utilities/json-schema.ts

```diff
@@ -19,16 +19,20 @@ import {
   Value,
 } from '../models/interface';
 
-function _getEnumFromValue<E, T extends string>(v: json.JsonValue, e: E, d: T): T {
-  if (typeof v !== 'string') {
-    return d;
+function _getEnumFromValue<E, T extends E[keyof E]>(
+  value: json.JsonValue,
+  enumeration: E,
+  defaultValue: T,
+): T {
+  if (typeof value !== 'string') {
+    return defaultValue;
   }
 
-  if (Object.values(e).indexOf(v) !== -1) {
-    return v as T;
+  if (Object.values(enumeration).indexOf(value) !== -1) {
+    return value as unknown as T;
```

Comment:
is this really the long-term fix? if not needs a TODO

Reply:
This is pretty permanent as type `T` doesn't provide enough type sufficiently overlaps for type `string`